BepInEx.Core/Bootstrap/BaseChainloader.cs

```diff
@@ -387,19 +414,26 @@ private void LoadPlugins(List<PluginInfo> plugins, bool provider)
                     var type = ass.GetType(plugin.TypeName);
                     plugin.Instance = Activator.CreateInstance(type);
                     AppDomain.CurrentDomain.AssemblyResolve += (_, args) => ((BasePluginProvider)plugin.Instance).ResolveAssembly(args.Name);
-                    var requestedPluginsStream = ((BasePluginProvider)plugin.Instance).GetPlugins();
-                    Loaders.AddRange(requestedPluginsStream);
+                    var pluginLoadContexts = ((BasePluginProvider)plugin.Instance).GetPlugins();
+                    foreach (IPluginLoadContext context in pluginLoadContexts)
+                    {
+                        LoadContexts[context] = plugin;
```

Comment:
Since the `plugin` local can now refer to either an actual plugin or a provider, I'd rename it to something like pluginInfo, same with other locals.